docs/reference/pip_install.rst

```diff
@@ -21,9 +21,52 @@ Overview
 
 Pip install has several stages:
 
-1. Resolve dependencies. What will be installed is determined here.
-2. Build wheels. All the dependencies that can be are built into wheels.
-3. Install the packages (and uninstall anything being upgraded/replaced).
+1. Identify the base requirements. The user supplied arguments are processed
+   here.
+2. Resolve dependencies. What will be installed is determined here.
+3. Build wheels. All the dependencies that can be are built into wheels.
```

Comment:
Well ```--no-binary``` doesn't (just) disable building wheels, it also stops consideration of wheels as candidates.

I'm inclined not to add this here - it should be a very high level summary of the order in which things happen (IMO). Hopefully (I haven't checked!) these options are already covered well in the more detailed part of the document.